keyvault/client/src/main/java/com/azure/keyvault/SecretClient.java

```diff
@@ -0,0 +1,540 @@
+import reactor.util.annotation.NonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+public final class SecretClient extends ServiceClient {
+    static final String SDK_NAME = "Azure-Keyvault";
+    static final String SDK_VERSION = "1.0.0-SNAPSHOT";
+    static final String API_VERSION = "7.0";
+    static final String ACCEPT_LANGUAGE = "en-US";
+    static final int DEFAULT_MAX_PAGE_RESULTS = 25;
+
+    private String vaultEndPoint;
+    private final SecretService service;
+
+
+    /**
+     * Creates a SecretClient that uses {@code credentials} to authorize with Azure and {@code pipeline} to
+     * service requests
+     *
+     * @param vaultEndPoint URL for the Application configuration service.
+     * @param pipeline HttpPipeline that the HTTP requests and responses flow through.
+     */
+    private SecretClient(String vaultEndPoint, HttpPipeline pipeline) {
+        super(pipeline);
+        this.vaultEndPoint = vaultEndPoint;
+        this.service = RestProxy.create(SecretService.class, this);
+    }
+
+    /**
+     * Creates a builder that can configure options for the SecretClient before creating an instance of it.
+     * @return A new Builder to create a SecretClient from.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Provides configuration options for instances of {@link SecretClient}.
+     */
+    public static final class Builder {
+        private final List<HttpPipelinePolicy> policies;
+        private ServiceClientCredentials credentials;
+        private String vaultEndPoint;
+        private HttpClient httpClient;
+        private HttpLogDetailLevel httpLogDetailLevel;
+        private RetryPolicy retryPolicy;
+        private String userAgent;
+
+        private Builder() {
+            userAgent = String.format("Azure-SDK-For-Java/%s (%s)", SDK_NAME, SDK_VERSION);
+            retryPolicy = new RetryPolicy();
+            httpLogDetailLevel = HttpLogDetailLevel.NONE;
+            policies = new ArrayList<>();
+        }
+
+        /**
+         * Creates a {@link SecretClient} based on options set in the Builder.
+         *
+         * Every time {@code build()} is called, a new instance of {@link SecretClient} is created.
+         *
+         * @return A SecretClient with the options set from the builder.
+         * @throws IllegalStateException If {@link Builder#credentials(ServiceClientCredentials)}
+         * @throws IllegalStateException If {@link Builder#credentials(ServiceClientCredentials)}
+         * has not been set.
+         */
+        public SecretClient build() {
+            if (credentials == null) {
+                throw new IllegalStateException("'credentials' is required.");
+            }
+
+            if (vaultEndPoint == null){
+                throw new IllegalStateException("'Vault's Endpoint Url' is required.");
+            }
```

Comment:
It might be nice to extract these error strings out into a single class, so that we may improve them over time in a centralised way. In time we might even do this across client libraries.

Reply:
@JonathanGiles Is there a way to fetch localised resource strings similar to .NET where they have a .resx that, based on the culture of the VM, can get a culture-specific string.

Reply:
Yes, you can externalise strings to properties files. This is a very common approach in Java, however I don't believe there is any intention to localise these strings. We should double check this though - I'll chase that up though, just to be certain.

Reply:
Extracted to KeyVaultErrorCodeStrings class.